[🐞 Bug] 大量数据（50万+）导出时UI会卡死最终结果无法导出

### 数据库类型和版本

MySQL  5.7.23

### 问题描述

根据SQL查询出结果后（数据超50万），点击有右下角导出按钮，“导出为EXCEL”，导出页面会卡死，进度长时间停留在0%，

### 错误日志

```shell
DBX debug log
Exported: 2026-06-17T02:40:01.689Z
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/149.0.0.0 Safari/537.36 Edg/149.0.0.0
Platform: Win32
Timezone: Asia/Shanghai

[2026-06-17T02:38:17.065Z] [INFO] [DBX][debug-log] enabled {"url":"http://tauri.localhost/","viewport":"1920x1032","devicePixelRatio":1,"language":"zh-CN","online":true}
[2026-06-17T02:38:17.065Z] [DEBUG] [DBX][api:start] saveDesktopSettings
[2026-06-17T02:38:17.098Z] [DEBUG] [DBX][api:success] {"operation":"saveDesktopSettings","elapsedMs":32}
[2026-06-17T02:38:24.514Z] [INFO] [DBX][DataGrid:load:start]
[2026-06-17T02:38:24.515Z] [INFO] [DBX][DataGrid:load:done] {"elapsed":"1ms"}
[2026-06-17T02:38:25.009Z] [DEBUG] [DBX][api:start] analyzeSqlReferences
[2026-06-17T02:38:25.013Z] [ERROR] [DBX][api:error] {"operation":"analyzeSqlReferences","elapsedMs":4,"error":"sql parser error: Expected: end of statement, found: select at Line: 7, Column: 1"}
[2026-06-17T02:38:28.730Z] [INFO] [DBX][DataGrid:load:start]
[2026-06-17T02:38:28.730Z] [INFO] [DBX][DataGrid:load:done] {"elapsed":"1ms"}
[2026-06-17T02:38:29.225Z] [DEBUG] [DBX][api:start] analyzeSqlReferences
[2026-06-17T02:38:29.231Z] [DEBUG] [DBX][api:success] {"operation":"analyzeSqlReferences","elapsedMs":6}
[2026-06-17T02:38:29.231Z] [DEBUG] [DBX][api:start] connectDb
[2026-06-17T02:38:29.571Z] [DEBUG] [DBX][api:success] {"operation":"connectDb","elapsedMs":340}
[2026-06-17T02:38:29.574Z] [DEBUG] [DBX][api:start] getColumns
[2026-06-17T02:38:30.985Z] [DEBUG] [DBX][api:start] listTables
[2026-06-17T02:38:31.462Z] [DEBUG] [DBX][api:success] {"operation":"getColumns","elapsedMs":1888}
[2026-06-17T02:38:31.778Z] [DEBUG] [DBX][api:success] {"operation":"listTables","elapsedMs":793}
[2026-06-17T02:38:31.778Z] [DEBUG] [DBX][api:start] listTables
[2026-06-17T02:38:32.176Z] [DEBUG] [DBX][api:success] {"operation":"listTables","elapsedMs":398}
[2026-06-17T02:38:32.177Z] [DEBUG] [DBX][api:start] getColumns
[2026-06-17T02:38:32.224Z] [DEBUG] [DBX][api:success] {"operation":"getColumns","elapsedMs":47}
[2026-06-17T02:38:32.224Z] [DEBUG] [DBX][api:start] getColumns
[2026-06-17T02:38:32.268Z] [DEBUG] [DBX][api:success] {"operation":"getColumns","elapsedMs":44}
[2026-06-17T02:38:32.268Z] [DEBUG] [DBX][api:start] getColumns
[2026-06-17T02:38:32.310Z] [DEBUG] [DBX][api:success] {"operation":"getColumns","elapsedMs":42}
[2026-06-17T02:38:35.131Z] [DEBUG] [ui:click] {"tag":"button"}
[2026-06-17T02:38:35.132Z] [INFO] [DBX][executeTabSql:start] {"traceId":"5330e874","tabId":"317b152f-17a7-4054-8278-37f19261bc8b","mode":"query","connectionId":"f51d48e8-86c8-4268-8a69-ee077deb8d18","database":"wx_product","sql":"SELECT ps.productsaleid FROM product_sale ps\nINNER JOIN product p ON ps.product=p.id\nINNER JOIN pss_total_stock pts ON pts.productsale=ps.productsaleid AND ps.merchant_id=\"202100010001\"\nWHERE p.mccategory IN (\n\"MC110103\",\"MC110105\",\"MC110106\",\"MC110201\",\"MC110202\",\"MC110203\",\"MC110204\",\"MC110803\",\"MC110805\",\"MC1214010\",\"MC1214020\",\"MC130101\",\"MC130102\",\"MC130104\",\"MC130301\",\"MC130303\",\"MC130305\",\"MC130307\",\"MC130310\",\"MC130311\",\"MC130401\",\"MC130402\",\"MC130403\",\"MC130501\",\"MC130502\",\"MC130505\",\"MC130506\",\"MC110101\",\"MC110501\",\"MC110503\",\"MC110506\",\"MC130601\",\"MC130602\",\"MC130603\",\"MC130605\",\"MC130606\",\"MC130607\",\"MC130701\",\"MC130801\",\"MC130802\",\"MC130902\",\"MC131001\",\"MC131002\",\"MC131003\",\"MC131004\",\"MC131101\",\"MC131102\",\"MC131105\",\"MC110601\",\"MC110609\",\"MC110610\",\"MC110701\",\"MC110708\",\"MC110901\",\"MC110902\",\"MC110916\",\"MC110917\",\"MC140101\",\"MC140301\",\"MC140501\",\"MC140602\",\"MC140606\",\"MC140608\",\"MC140705\",\"MC140711\",\"MC140801\",\"MC140805\",\"MC141301\",\"MC141404\",\"MC141405\",\"MC141601\",\"MC141603\",\"MC141604\",\"MC141701\",\"MC141702\",\"MC141704\",\"MC141801\",\"MC141802\",\"MC141805\",\"MC141809\",\"MC141813\",\"MC160205\",\"MC150102\",\"MC150107\",\"MC150201\",\"MC150203\",\"MC150204\",\"MC150206\",\"MC1204010\",\"MC1204020\",\"MC1204030\",\"MC1205020\",\"MC1217020\",\"MC1218010\",\"MC150101\",\"CBMC00001\",\"CBMC00002\",\"CBMC00003\",\"CBMC00004\",\"MC110102\",\"MC110104\",\"MC110301\",\"MC110302\",\"MC110303\",\"MC110401\",\"MC110402\",\"MC110502\",\"MC110504\",\"MC110505\",\"MC110507\",\"MC110602\",\"MC110603\",\"MC110604\",\"MC110605\",\"MC110606\",\"MC110607\",\"MC110608\",\"MC110702\",\"MC110703\",\"MC110704\",\"MC110705\",\"MC110706\",\"MC110707\",\"MC110709\",\"MC110710\",\"MC110711\",\"MC110712\",\"MC110801\",\"MC110802\",\"MC110804\",\"MC110806\",\"MC110903\",\"MC110904\",\"MC110905\",\"MC110906\",\"MC110907\",\"MC110908\",\"MC110909\",\"MC110910\",\"MC110911\",\"MC110912\",\"MC110913\",\"MC110914\",\"MC110915\",\"MC120101\",\"MC1201011\",\"MC1201012\",\"MC120102\",\"MC1201021\",\"MC1201022\",\"MC120103\",\"MC1201031\",\"MC1201032\",\"MC120104\",\"MC1201041\",\"MC1201042\",\"MC1201043\",\"MC1201050\",\"MC1201060\",\"MC1201070\",\"MC120201\",\"MC1202011\",\"MC1202012\",\"MC120202\",\"MC1202021\",\"MC1202022\",\"MC120203\",\"MC1202031\",\"MC1202032\",\"MC120204\",\"MC1202041\",\"MC1202042\",\"MC120205\",\"MC1202051\",\"MC1202052\",\"MC120206\",\"MC1202061\",\"MC1202062\",\"MC1202063\",\"MC1202070\",\"MC1202080\",\"MC1202090\",\"MC1202091\",\"MC120301\",\"MC1203011\",\"MC1203012\",\"MC120302\",\"MC1203021\",\"MC1203022\",\"MC120303\",\"MC1203031\",\"MC1203032\",\"MC120304\",\"MC1203041\",\"MC1203042\",\"MC120305\",\"MC1203051\",\"MC1203052\",\"MC120306\",\"MC1203061\",\"MC1203062\",\"MC1203063\",\"MC1203070\",\"MC1203080\",\"MC1203090\",\"MC120401\",\"MC120402\",\"MC120403\",\"MC120501\",\"MC1205010\",\"MC120502\",\"MC120503\",\"MC1205030\",\"MC120601\",\"MC1206010\",\"MC120602\",\"MC1206020\",\"MC120603\",\"MC1206030\",\"MC120604\",\"MC120701\",\"MC1207010\",\"MC120702\",\"MC1207020\",\"MC120703\",\"MC1207030\",\"MC120704\",\"MC120801\",\"MC1208010\",\"MC120802\",\"MC120803\",\"MC120901\",\"MC1209011\",\"MC1209012\",\"MC120902\",\"MC1209021\",\"MC1209022\",\"MC1209023\",\"MC1209024\",\"MC1209025\",\"MC1209026\",\"MC120903\",\"MC1209031\",\"MC1209032\",\"MC1209033\",\"MC1209034\",\"MC1209035\",\"MC1209036\",\"MC120904\",\"MC120905\",\"MC120906\",\"MC120907\",\"MC120908\",\"MC120909\",\"MC120910\",\"MC120911\",\"MC121001\",\"MC1210010\",\"MC121002\",\"MC1210020\",\"MC1210030\",\"MC1211000\",\"MC121101\",\"MC121102\",\"MC1211100\",\"MC121201\",\"MC1212010\",\"MC121202\",\"MC1212020\",\"MC...
[2026-06-17T02:38:35.136Z] [DEBUG] [DBX][api:start] prepareQueryPaginationExecutionPlan
[2026-06-17T02:38:35.145Z] [DEBUG] [DBX][api:success] {"operation":"prepareQueryPaginationExecutionPlan","elapsedMs":9}
[2026-06-17T02:38:35.145Z] [INFO] [DBX][executeTabSql:execute-multi:start] {"traceId":"5330e874","elapsed":"14ms"}
[2026-06-17T02:38:35.145Z] [DEBUG] [DBX][api:start] executeMulti
[2026-06-17T02:38:40.058Z] [DEBUG] [DBX][api:start] analyzeSqlReferences
[2026-06-17T02:38:40.059Z] [DEBUG] [DBX][api:start] listTables
[2026-06-17T02:38:40.079Z] [DEBUG] [DBX][api:success] {"operation":"analyzeSqlReferences","elapsedMs":22}
[2026-06-17T02:38:40.080Z] [DEBUG] [DBX][api:start] getColumns
[2026-06-17T02:38:40.080Z] [DEBUG] [DBX][api:start] getColumns
[2026-06-17T02:38:40.080Z] [DEBUG] [DBX][api:start] getColumns
[2026-06-17T02:38:40.101Z] [DEBUG] [DBX][api:success] {"operation":"listTables","elapsedMs":42}
[2026-06-17T02:38:40.101Z] [DEBUG] [DBX][api:start] listTables
[2026-06-17T02:38:40.142Z] [DEBUG] [DBX][api:success] {"operation":"listTables","elapsedMs":41}
[2026-06-17T02:38:40.156Z] [ERROR] [DBX][api:error] {"operation":"getColumns","elapsedMs":76,"error":"MySQL connection failed: Server error: `ERROR 42000 (1044): Access denied for user 'wxzx_jumpserver'@'10.1.5.240' to database 'product''"}
[2026-06-17T02:38:40.156Z] [ERROR] [DBX][api:error] {"operation":"getColumns","elapsedMs":76,"error":"MySQL connection failed: Server error: `ERROR 42000 (1044): Access denied for user 'wxzx_jumpserver'@'10.1.5.240' to database 'pss_total_stock''"}
[2026-06-17T02:38:40.156Z] [ERROR] [DBX][api:error] {"operation":"getColumns","elapsedMs":76,"error":"MySQL connection failed: Server error: `ERROR 42000 (1044): Access denied for user 'wxzx_jumpserver'@'10.1.5.240' to database 'product_sale''"}
[2026-06-17T02:38:40.992Z] [DEBUG] [DBX][api:start] listTables
[2026-06-17T02:38:41.028Z] [DEBUG] [DBX][api:success] {"operation":"listTables","elapsedMs":36}
[2026-06-17T02:38:41.028Z] [DEBUG] [DBX][api:start] listTables
[2026-06-17T02:38:41.069Z] [DEBUG] [DBX][api:success] {"operation":"listTables","elapsedMs":41}
[2026-06-17T02:38:42.030Z] [DEBUG] [DBX][api:success] {"operation":"executeMulti","elapsedMs":6885}
[2026-06-17T02:38:42.030Z] [INFO] [DBX][executeTabSql:execute-multi:done] {"traceId":"5330e874","resultCount":1,"rowCounts":[100000],"columnCounts":[1],"elapsed":"6899ms"}
[2026-06-17T02:38:42.031Z] [INFO] [DBX][executeTabSql:count:start] {"traceId":"5330e874","elapsed":"6899ms"}
[2026-06-17T02:38:42.031Z] [DEBUG] [DBX][api:start] executeQuery
[2026-06-17T02:38:42.031Z] [INFO] [DBX][executeTabSql:result:assigned] {"traceId":"5330e874","rowCount":100000,"columnCount":1,"backendMs":6841,"elapsed":"6900ms"}
[2026-06-17T02:38:42.031Z] [INFO] [DBX][executeTabSql:metadata:start] {"traceId":"5330e874","elapsed":"6900ms"}
[2026-06-17T02:38:42.031Z] [INFO] [DBX][executeTabSql:metadata:editability:start] {"traceId":"5330e874","elapsed":"6900ms"}
[2026-06-17T02:38:42.032Z] [DEBUG] [DBX][api:start] analyzeEditableQueryEditability
[2026-06-17T02:38:42.032Z] [INFO] [DBX][executeTabSql:finish] {"traceId":"5330e874","elapsed":"6900ms"}
[2026-06-17T02:38:42.033Z] [INFO] [DBX][ContentArea:result:observed] {"tabId":"317b152f-17a7-4054-8278-37f19261bc8b","rowCount":100000,"columnCount":1,"backendMs":6841,"isExecuting":false}
[2026-06-17T02:38:42.042Z] [INFO] [DBX][DataGrid:setup] {"traceId":"6393eb62","cacheKey":"317b152f-17a7-4054-8278-37f19261bc8b-606c1c6e-9a5e-4b83-89c9-ad9df7f055c5-0","rowCount":100000,"columnCount":1,"backendMs":6841,"loading":false}
[2026-06-17T02:38:42.043Z] [INFO] [DBX][DataGrid:result:prop] {"traceId":"6393eb62","cacheKey":"317b152f-17a7-4054-8278-37f19261bc8b-606c1c6e-9a5e-4b83-89c9-ad9df7f055c5-0","rowCount":100000,"columnCount":1,"backendMs":6841,"loading":false,"elapsedSinceSetup":"1ms"}
[2026-06-17T02:38:42.126Z] [INFO] [DBX][DataGrid:display-items:ready] {"traceId":"6393eb62","cacheKey":"317b152f-17a7-4054-8278-37f19261bc8b-606c1c6e-9a5e-4b83-89c9-ad9df7f055c5-0","displayItemCount":100000,"sourceRowCount":100000,"elapsedSinceSetup":"84ms"}
[2026-06-17T02:38:42.243Z] [INFO] [DBX][ContentArea:result:nextTick] {"tabId":"317b152f-17a7-4054-8278-37f19261bc8b","elapsed":"210ms","isExecuting":false}
[2026-06-17T02:38:42.243Z] [INFO] [DBX][DataGrid:result:nextTick] {"traceId":"6393eb62","cacheKey":"317b152f-17a7-4054-8278-37f19261bc8b-606c1c6e-9a5e-4b83-89c9-ad9df7f055c5-0","elapsed":"200ms","loading":false}
[2026-06-17T02:38:42.250Z] [DEBUG] [DBX][api:start] saveHistory
[2026-06-17T02:38:42.255Z] [INFO] [DBX][ContentArea:result:first-frame] {"tabId":"317b152f-17a7-4054-8278-37f19261bc8b","elapsed":"222ms","isExecuting":false}
[2026-06-17T02:38:42.256Z] [INFO] [DBX][DataGrid:result:first-frame] {"traceId":"6393eb62","cacheKey":"317b152f-17a7-4054-8278-37f19261bc8b-606c1c6e-9a5e-4b83-89c9-ad9df7f055c5-0","elapsed":"213ms","loading":false}
[2026-06-17T02:38:42.256Z] [INFO] [DBX][DataGrid:display-items:first-frame] {"traceId":"6393eb62","cacheKey":"317b152f-17a7-4054-8278-37f19261bc8b-606c1c6e-9a5e-4b83-89c9-ad9df7f055c5-0","displayItemCount":100000,"renderedRows":30,"elapsed":"130ms","elapsedSinceSetup":"214ms","loading":false}
[2026-06-17T02:38:42.271Z] [DEBUG] [DBX][api:success] {"operation":"analyzeEditableQueryEditability","elapsedMs":239}
[2026-06-17T02:38:42.271Z] [INFO] [DBX][executeTabSql:metadata:editability:done] {"traceId":"5330e874","editable":false,"reason":"complex-source","elapsed":"7139ms"}
[2026-06-17T02:38:42.271Z] [INFO] [DBX][executeTabSql:metadata:done] {"traceId":"5330e874","elapsed":"7140ms"}
[2026-06-17T02:38:42.273Z] [DEBUG] [DBX][api:success] {"operation":"saveHistory","elapsedMs":24}
[2026-06-17T02:38:44.457Z] [DEBUG] [DBX][api:start] listTables
[2026-06-17T02:38:44.530Z] [DEBUG] [DBX][api:success] {"operation":"listTables","elapsedMs":73}
[2026-06-17T02:38:44.531Z] [DEBUG] [DBX][api:start] listTables
[2026-06-17T02:38:44.609Z] [DEBUG] [DBX][api:start] listTables
[2026-06-17T02:38:44.640Z] [DEBUG] [DBX][api:start] listTables
[2026-06-17T02:38:44.908Z] [DEBUG] [DBX][api:success] {"operation":"listTables","elapsedMs":298}
[2026-06-17T02:38:44.908Z] [DEBUG] [DBX][api:start] listTables
[2026-06-17T02:38:45.016Z] [DEBUG] [DBX][api:success] {"operation":"listTables","elapsedMs":376}
[2026-06-17T02:38:45.016Z] [DEBUG] [DBX][api:start] listTables
[2026-06-17T02:38:45.245Z] [DEBUG] [DBX][api:success] {"operation":"listTables","elapsedMs":337}
[2026-06-17T02:38:45.370Z] [DEBUG] [DBX][api:success] {"operation":"listTables","elapsedMs":354}
[2026-06-17T02:38:45.561Z] [DEBUG] [DBX][api:success] {"operation":"listTables","elapsedMs":1030}
[2026-06-17T02:38:52.742Z] [DEBUG] [ui:click] {"tag":"button","role":"menuitem","type":"button","label":"导出 XLSX"}
[2026-06-17T02:38:56.406Z] [DEBUG] [DBX][api:start] prepareQueryPaginationExecutionPlan
[2026-06-17T02:38:56.419Z] [DEBUG] [DBX][api:success] {"operation":"prepareQueryPaginationExecutionPlan","elapsedMs":13}
[2026-06-17T02:38:56.420Z] [DEBUG] [DBX][api:start] executeMulti
[2026-06-17T02:39:03.507Z] [DEBUG] [DBX][api:success] {"operation":"executeMulti","elapsedMs":7087}
[2026-06-17T02:39:03.519Z] [DEBUG] [DBX][api:start] prepareQueryPaginationExecutionPlan
[2026-06-17T02:39:03.563Z] [DEBUG] [DBX][api:success] {"operation":"prepareQueryPaginationExecutionPlan","elapsedMs":44}
[2026-06-17T02:39:03.563Z] [DEBUG] [DBX][api:start] executeMulti
[2026-06-17T02:39:12.261Z] [ERROR] [DBX][api:error] {"operation":"executeQuery","elapsedMs":30230,"error":"Query timed out after 30 seconds"}
[2026-06-17T02:39:12.262Z] [WARN] [DBX][executeTabSql:count:error] {"traceId":"5330e874","elapsed":"37131ms","error":"Query timed out after 30 seconds"}
[2026-06-17T02:39:19.747Z] [DEBUG] [DBX][api:success] {"operation":"executeMulti","elapsedMs":16184}
[2026-06-17T02:39:19.752Z] [DEBUG] [DBX][api:start] prepareQueryPaginationExecutionPlan
[2026-06-17T02:39:19.762Z] [DEBUG] [DBX][api:success] {"operation":"prepareQueryPaginationExecutionPlan","elapsedMs":11}
[2026-06-17T02:39:19.763Z] [DEBUG] [DBX][api:start] executeMulti
[2026-06-17T02:39:39.569Z] [INFO] [document:visibilitychange] hidden
[2026-06-17T02:39:43.680Z] [LOG] [STARTUP] frontend bootstrap begin
[2026-06-17T02:39:43.758Z] [LOG] [STARTUP] frontend modules loaded
[2026-06-17T02:39:43.765Z] [LOG] [STARTUP] locale ready
[2026-06-17T02:39:43.835Z] [LOG] [STARTUP] onMounted begin
[2026-06-17T02:39:43.856Z] [LOG] [STARTUP] initApp begin
[2026-06-17T02:39:43.857Z] [DEBUG] [DBX][api:start] loadDesktopSettings
[2026-06-17T02:39:43.857Z] [DEBUG] [DBX][api:start] loadAiConfig
[2026-06-17T02:39:43.858Z] [DEBUG] [DBX][api:start] getAppVersion
[2026-06-17T02:39:43.859Z] [DEBUG] [DBX][api:start] pendingOpenSqlFiles
[2026-06-17T02:39:43.859Z] [DEBUG] [DBX][api:start] pendingOpenDbFiles
[2026-06-17T02:39:43.860Z] [DEBUG] [DBX][api:start] pendingOpenConnectionLinks
[2026-06-17T02:39:43.860Z] [LOG] [STARTUP] onMounted sync done: 24ms
[2026-06-17T02:39:43.860Z] [LOG] [STARTUP] vue mounted
[2026-06-17T02:39:43.994Z] [DEBUG] [DBX][api:success] {"operation":"pendingOpenConnectionLinks","elapsedMs":135}
[2026-06-17T02:39:43.994Z] [DEBUG] [DBX][api:success] {"operation":"loadDesktopSettings","elapsedMs":138}
[2026-06-17T02:39:43.995Z] [INFO] [DBX][debug-log] enabled {"url":"http://tauri.localhost/","viewport":"1920x1032","devicePixelRatio":1,"language":"zh-CN","online":true}
[2026-06-17T02:39:43.995Z] [DEBUG] [DBX][api:start] loadSavedSqlLibrary
[2026-06-17T02:39:43.997Z] [DEBUG] [DBX][api:success] {"operation":"pendingOpenSqlFiles","elapsedMs":138}
[2026-06-17T02:39:43.997Z] [DEBUG] [DBX][api:success] {"operation":"pendingOpenDbFiles","elapsedMs":138}
[2026-06-17T02:39:43.997Z] [DEBUG] [DBX][api:success] {"operation":"loadAiConfig","elapsedMs":140}
[2026-06-17T02:39:44.004Z] [DEBUG] [DBX][api:success] {"operation":"getAppVersion","elapsedMs":146}
[2026-06-17T02:39:44.006Z] [DEBUG] [DBX][api:success] {"operation":"loadSavedSqlLibrary","elapsedMs":11}
[2026-06-17T02:39:44.007Z] [LOG] [STARTUP]   savedSqlStore.initFromStorage: 151ms
[2026-06-17T02:39:44.007Z] [DEBUG] [DBX][api:start] loadPinnedTreeNodeIds
[2026-06-17T02:39:44.012Z] [DEBUG] [DBX][api:success] {"operation":"loadPinnedTreeNodeIds","elapsedMs":5}
[2026-06-17T02:39:44.013Z] [DEBUG] [DBX][api:start] loadConnections
[2026-06-17T02:39:44.024Z] [INFO] [DBX][DataGrid:load:start]
[2026-06-17T02:39:44.060Z] [DEBUG] [DBX][api:success] {"operation":"loadConnections","elapsedMs":47}
[2026-06-17T02:39:44.060Z] [DEBUG] [DBX][api:start] loadSidebarLayout
[2026-06-17T02:39:44.069Z] [INFO] [DBX][DataGrid:load:done] {"elapsed":"45ms"}
[2026-06-17T02:39:44.079Z] [DEBUG] [DBX][api:success] {"operation":"loadSidebarLayout","elapsedMs":18}
[2026-06-17T02:39:44.371Z] [LOG] [STARTUP]   connectionStore.initFromDisk: 515ms
[2026-06-17T02:39:44.382Z] [WARN] [performance:long-task] {"name":"self","durationMs":293,"startTimeMs":569}
[2026-06-17T02:39:44.451Z] [DEBUG] [DBX][api:start] analyzeSqlReferences
[2026-06-17T02:39:44.469Z] [DEBUG] [DBX][api:success] {"operation":"analyzeSqlReferences","elapsedMs":18}
[2026-06-17T02:39:44.470Z] [DEBUG] [DBX][api:start] connectDb
[2026-06-17T02:39:44.470Z] [DEBUG] [DBX][api:start] connectDb
[2026-06-17T02:39:44.470Z] [DEBUG] [DBX][api:start] connectDb
[2026-06-17T02:39:44.746Z] [DEBUG] [DBX][api:success] {"operation":"connectDb","elapsedMs":276}
[2026-06-17T02:39:44.747Z] [DEBUG] [DBX][api:start] getColumns
[2026-06-17T02:39:44.773Z] [DEBUG] [DBX][api:success] {"operation":"connectDb","elapsedMs":304}
[2026-06-17T02:39:44.774Z] [DEBUG] [DBX][api:start] getColumns
[2026-06-17T02:39:44.846Z] [ERROR] [DBX][api:error] {"operation":"getColumns","elapsedMs":99,"error":"MySQL connection failed: Server error: `ERROR 42000 (1044): Access denied for user 'wxzx_jumpserver'@'10.1.5.240' to database 'pss_total_stock''"}
[2026-06-17T02:39:44.852Z] [ERROR] [DBX][api:error] {"operation":"getColumns","elapsedMs":78,"error":"MySQL connection failed: Server error: `ERROR 42000 (1044): Access denied for user 'wxzx_jumpserver'@'10.1.5.240' to database 'product_sale''"}
[2026-06-17T02:39:44.989Z] [DEBUG] [DBX][api:success] {"operation":"connectDb","elapsedMs":519}
[2026-06-17T02:39:44.989Z] [DEBUG] [DBX][api:start] getColumns
[2026-06-17T02:39:45.086Z] [ERROR] [DBX][api:error] {"operation":"getColumns","elapsedMs":97,"error":"MySQL connection failed: Server error: `ERROR 42000 (1044): Access denied for user 'wxzx_jumpserver'@'10.1.5.240' to database 'product''"}
[2026-06-17T02:39:45.865Z] [DEBUG] [ui:click] {"tag":"button"}
[2026-06-17T02:39:45.877Z] [DEBUG] [DBX][api:start] listSystemFonts
[2026-06-17T02:39:45.934Z] [DEBUG] [DBX][api:success] {"operation":"listSystemFonts","elapsedMs":57}
[2026-06-17T02:39:47.643Z] [DEBUG] [ui:click] {"tag":"button","type":"button","label":"外观"}
[2026-06-17T02:39:49.328Z] [DEBUG] [ui:click] {"tag":"button","type":"button","label":"导航"}
[2026-06-17T02:39:49.915Z] [DEBUG] [ui:click] {"tag":"button","type":"button","label":"SQL 格式化"}
[2026-06-17T02:39:51.218Z] [DEBUG] [ui:click] {"tag":"button","type":"button","label":"数据"}
[2026-06-17T02:39:51.811Z] [DEBUG] [ui:click] {"tag":"button","type":"button","label":"Redis"}
[2026-06-17T02:39:52.261Z] [DEBUG] [ui:click] {"tag":"button","type":"button","label":"快捷键"}
[2026-06-17T02:39:53.138Z] [DEBUG] [ui:click] {"tag":"button","type":"button","label":"代码片段"}
[2026-06-17T02:39:53.859Z] [DEBUG] [DBX][api:start] checkForUpdates
[2026-06-17T02:39:55.830Z] [DEBUG] [DBX][api:success] {"operation":"checkForUpdates","elapsedMs":1971}
[2026-06-17T02:39:55.974Z] [DEBUG] [ui:click] {"tag":"button","type":"button","label":"编辑器"}
[2026-06-17T02:39:57.623Z] [DEBUG] [ui:click] {"tag":"button","type":"button","label":"外观"}
[2026-06-17T02:40:01.689Z] [DEBUG] [ui:click] {"tag":"button","type":"button","label":"下载日志"}

===== Native / Tauri logs =====
Native log dir: C:\Users\HP\AppData\Local\com.dbx.app\logs

===== DBX.log =====
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC160515"), span: Span(Location(5,5398)..Location(5,5408)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5408)..Location(5,5409)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC160516"), span: Span(Location(5,5409)..Location(5,5419)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5419)..Location(5,5420)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC160517"), span: Span(Location(5,5420)..Location(5,5430)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5430)..Location(5,5431)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC160518"), span: Span(Location(5,5431)..Location(5,5441)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5441)..Location(5,5442)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC160519"), span: Span(Location(5,5442)..Location(5,5452)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5452)..Location(5,5453)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1201010"), span: Span(Location(5,5453)..Location(5,5464)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5464)..Location(5,5465)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1202010"), span: Span(Location(5,5465)..Location(5,5476)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5476)..Location(5,5477)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1203010"), span: Span(Location(5,5477)..Location(5,5488)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5488)..Location(5,5489)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1180010"), span: Span(Location(5,5489)..Location(5,5500)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5500)..Location(5,5501)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1280010"), span: Span(Location(5,5501)..Location(5,5512)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5512)..Location(5,5513)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1380010"), span: Span(Location(5,5513)..Location(5,5524)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5524)..Location(5,5525)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1206011"), span: Span(Location(5,5525)..Location(5,5536)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5536)..Location(5,5537)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1209013"), span: Span(Location(5,5537)..Location(5,5548)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5548)..Location(5,5549)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1210011"), span: Span(Location(5,5549)..Location(5,5560)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5560)..Location(5,5561)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1210012"), span: Span(Location(5,5561)..Location(5,5572)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5572)..Location(5,5573)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1212011"), span: Span(Location(5,5573)..Location(5,5584)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5584)..Location(5,5585)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1213030"), span: Span(Location(5,5585)..Location(5,5596)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5596)..Location(5,5597)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1213037"), span: Span(Location(5,5597)..Location(5,5608)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5608)..Location(5,5609)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1213038"), span: Span(Location(5,5609)..Location(5,5620)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5620)..Location(5,5621)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1213039"), span: Span(Location(5,5621)..Location(5,5632)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5632)..Location(5,5633)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1213045"), span: Span(Location(5,5633)..Location(5,5644)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5644)..Location(5,5645)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1213120"), span: Span(Location(5,5645)..Location(5,5656)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5656)..Location(5,5657)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1216040"), span: Span(Location(5,5657)..Location(5,5668)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5668)..Location(5,5669)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1218011"), span: Span(Location(5,5669)..Location(5,5680)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5680)..Location(5,5681)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1218012"), span: Span(Location(5,5681)..Location(5,5692)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5692)..Location(5,5693)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1218013"), span: Span(Location(5,5693)..Location(5,5704)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5704)..Location(5,5705)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1580010"), span: Span(Location(5,5705)..Location(5,5716)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Comma, span: Span(Location(5,5716)..Location(5,5717)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: DoubleQuotedString("MC1480010"), span: Span(Location(5,5717)..Location(5,5728)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: RParen, span: Span(Location(6,1)..Location(6,2)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Word(Word { value: "AND", quote_style: None, keyword: AND }), span: Span(Location(6,3)..Location(6,6)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 10
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] infix: TokenWithSpan { token: Word(Word { value: "AND", quote_style: None, keyword: AND }), span: Span(Location(6,3)..Location(6,6)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Identifier(Ident { value: "availablequantity", quote_style: None, span: Span(Location(6,11)..Location(6,28)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Minus, span: Span(Location(6,29)..Location(6,30)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 30
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: CompoundIdentifier([Ident { value: "pts", quote_style: None, span: Span(Location(6,7)..Location(6,10)) }, Ident { value: "availablequantity", quote_style: None, span: Span(Location(6,11)..Location(6,28)) }])
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Minus, span: Span(Location(6,29)..Location(6,30)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 30
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] infix: TokenWithSpan { token: Minus, span: Span(Location(6,29)..Location(6,30)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Identifier(Ident { value: "lockoccupy", quote_style: None, span: Span(Location(6,35)..Location(6,45)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Minus, span: Span(Location(6,46)..Location(6,47)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 30
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: CompoundIdentifier([Ident { value: "pts", quote_style: None, span: Span(Location(6,31)..Location(6,34)) }, Ident { value: "lockoccupy", quote_style: None, span: Span(Location(6,35)..Location(6,45)) }])
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Minus, span: Span(Location(6,46)..Location(6,47)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 30
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: Minus, span: Span(Location(6,46)..Location(6,47)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 30
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] infix: TokenWithSpan { token: Minus, span: Span(Location(6,46)..Location(6,47)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Identifier(Ident { value: "lock_available_quantity", quote_style: None, span: Span(Location(6,52)..Location(6,75)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: GtEq, span: Span(Location(6,76)..Location(6,78)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 20
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: CompoundIdentifier([Ident { value: "pts", quote_style: None, span: Span(Location(6,48)..Location(6,51)) }, Ident { value: "lock_available_quantity", quote_style: None, span: Span(Location(6,52)..Location(6,75)) }])
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: GtEq, span: Span(Location(6,76)..Location(6,78)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 20
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: GtEq, span: Span(Location(6,76)..Location(6,78)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 20
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] infix: TokenWithSpan { token: GtEq, span: Span(Location(6,76)..Location(6,78)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] parsing expr
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] prefix: Value(ValueWithSpan { value: Number("3", false), span: Span(Location(6,78)..Location(6,79)) })
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: SemiColon, span: Span(Location(6,80)..Location(6,81)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: SemiColon, span: Span(Location(6,80)..Location(6,81)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:44][sqlparser::dialect][DEBUG] get_next_precedence_full() TokenWithSpan { token: SemiColon, span: Span(Location(6,80)..Location(6,81)) }
[2026-06-17][02:39:44][sqlparser::parser][DEBUG] next precedence: 0
[2026-06-17][02:39:53][reqwest::connect][DEBUG] starting new connection: https://dl.dbxio.com/
[2026-06-17][02:39:53][reqwest::connect][DEBUG] starting new connection: https://github.com/
[2026-06-17][02:39:53][hyper_util::client::legacy::connect::http][DEBUG] connecting to 104.21.87.191:443
[2026-06-17][02:39:53][hyper_util::client::legacy::connect::http][DEBUG] connecting to 20.205.243.166:443
[2026-06-17][02:39:53][reqwest::connect][DEBUG] starting new connection: https://dl.dbxio.com/
[2026-06-17][02:39:53][reqwest::connect][DEBUG] starting new connection: https://github.com/
[2026-06-17][02:39:53][hyper_util::client::legacy::connect::http][DEBUG] connecting to 104.21.87.191:443
[2026-06-17][02:39:53][hyper_util::client::legacy::connect::http][DEBUG] connecting to 20.205.243.166:443
[2026-06-17][02:39:54][hyper_util::client::legacy::connect::http][DEBUG] connected to 20.205.243.166:443
[2026-06-17][02:39:54][hyper_util::client::legacy::connect::http][DEBUG] connected to 104.21.87.191:443
[2026-06-17][02:39:54][hyper_util::client::legacy::connect::http][DEBUG] connected to 20.205.243.166:443
[2026-06-17][02:39:54][hyper_util::client::legacy::connect::http][DEBUG] connected to 104.21.87.191:443
[2026-06-17][02:39:54][hyper_util::client::legacy::pool][DEBUG] pooling idle connection for ("https", github.com)
[2026-06-17][02:39:54][hyper_util::client::legacy::pool][DEBUG] reuse idle connection for ("https", github.com)
[2026-06-17][02:39:54][hyper_util::client::legacy::pool][DEBUG] pooling idle connection for ("https", github.com)
[2026-06-17][02:39:54][reqwest::connect][DEBUG] starting new connection: https://release-assets.githubusercontent.com/
[2026-06-17][02:39:54][hyper_util::client::legacy::connect::http][DEBUG] connecting to 185.199.109.133:443
[2026-06-17][02:39:54][hyper_util::client::legacy::connect::http][DEBUG] connected to 185.199.109.133:443
[2026-06-17][02:39:54][hyper_util::client::legacy::pool][DEBUG] pooling idle connection for ("https", dl.dbxio.com)
[2026-06-17][02:39:55][hyper_util::client::legacy::pool][DEBUG] pooling idle connection for ("https", dl.dbxio.com)
[2026-06-17][02:39:55][reqwest::connect][DEBUG] starting new connection: https://api.github.com/
[2026-06-17][02:39:55][hyper_util::client::legacy::connect::http][DEBUG] connecting to 20.205.243.168:443
[2026-06-17][02:39:55][hyper_util::client::legacy::connect::http][DEBUG] connected to 20.205.243.168:443
[2026-06-17][02:39:55][hyper_util::client::legacy::pool][DEBUG] pooling idle connection for ("https", api.github.com)
```

### DBX 版本

v0.5.33

### 操作系统

Windows

### 优先级（是否紧急）

🔴 P0 紧急 · 下个版本必须修

### 补充信息

<img width="1871" height="966" alt="Image" src="https://github.com/user-attachments/assets/fbb4ea24-24a8-48d8-ad3b-d325b322adc8" />

--- a/crates/dbx-core/assets/database-drivers.manifest.json
+++ b/crates/dbx-core/assets/database-drivers.manifest.json
@@ -1474,8 +1474,8 @@
       "supportLevel": "connect",
       "capabilities": {
         "queryExecution": false,
-        "metadataBrowse": false,
-        "objectBrowser": false,
+        "metadataBrowse": true,
+        "objectBrowser": true,
         "objectSource": false,
         "schemaSearch": false,
         "diagram": false,

--- a/crates/dbx-core/src/database_capabilities.rs
+++ b/crates/dbx-core/src/database_capabilities.rs
@@ -37,8 +37,14 @@ pub fn is_metadata_connection_scoped(db_type: &DatabaseType) -> bool {
 }
 
 pub fn skips_tcp_probe(db_type: &DatabaseType) -> bool {
-    matches!(db_type, DatabaseType::Sqlite | DatabaseType::DuckDb | DatabaseType::Turso | DatabaseType::Jdbc)
-        || is_agent_type(db_type)
+    matches!(
+        db_type,
+        DatabaseType::Sqlite
+            | DatabaseType::DuckDb
+            | DatabaseType::Turso
+            | DatabaseType::Jdbc
+            | DatabaseType::MessageQueue
+    ) || is_agent_type(db_type)
 }
 
 /// Database types whose connection backs onto a single local file (or may, in the

--- a/crates/dbx-core/src/table_export.rs
+++ b/crates/dbx-core/src/table_export.rs
@@ -402,7 +402,12 @@ pub async fn export_table_data_core(
                 error_message: None,
             });
 
-            finish_streaming_xlsx_workbook(writer).map_err(|e| format!("Failed to finalize XLSX file: {e}"))?;
+            // Explicitly flush the XLSX writer's BufWriter so IO errors
+            // (e.g. disk-full) are surfaced rather than silently swallowed
+            // by Drop.
+            let mut xlsx_buf =
+                finish_streaming_xlsx_workbook(writer).map_err(|e| format!("Failed to finalize XLSX file: {e}"))?;
+            xlsx_buf.flush().map_err(|e| format!("Failed to flush XLSX file: {e}"))?;
         }
         "json" => {
             file.write_all(b"[\n").map_err(|e| format!("Failed to write JSON: {e}"))?;

--- a/crates/dbx-core/src/table_export.rs
+++ b/crates/dbx-core/src/table_export.rs
@@ -10,7 +10,7 @@ use crate::transfer::{
     count_sql_with_where, execute_on_pool, execute_on_pool_with_max_rows, keyset_pagination_sql,
     pagination_sql_with_filter_order,
 };
-use crate::xlsx_export::{build_xlsx_workbook, XlsxWorksheetData};
+use crate::xlsx_export::{finish_streaming_xlsx_workbook, start_streaming_xlsx_workbook};
 
 const DEFAULT_BATCH_SIZE: usize = 10_000;
 
@@ -325,7 +325,13 @@ pub async fn export_table_data_core(
             }
         }
         "xlsx" => {
-            let mut all_rows: Vec<Vec<Value>> = Vec::new();
+            // Create a dedicated file handle for the streaming XLSX writer
+            // instead of cloning the outer BufWriter's handle.  This avoids
+            // sharing a file descriptor between two independent buffers.
+            let xlsx_file =
+                std::fs::File::create(&request.file_path).map_err(|e| format!("Failed to create XLSX file: {e}"))?;
+            let mut writer =
+                start_streaming_xlsx_workbook(BufWriter::new(xlsx_file), Some(&request.table_name), &col_names)?;
 
             loop {
                 // Check cancellation between batches
@@ -358,12 +364,14 @@ pub async fn export_table_data_core(
                     break;
                 }
 
-                all_rows.extend(result.rows);
+                for row in &result.rows {
+                    writer.write_row(row).map_err(|e| format!("Failed to write XLSX row: {e}"))?;
+                }
                 rows_exported += row_count as u64;
 
                 if use_keyset {
                     // Keyset pagination: track last PK values for next batch
-                    if let Some(last_row) = all_rows.last() {
+                    if let Some(last_row) = result.rows.last() {
                         last_pk_values = pk_indices.iter().map(|&i| last_row[i].clone()).collect();
                     }
                 } else {
@@ -394,11 +402,7 @@ pub async fn export_table_data_core(
                 error_message: None,
             });
 
-            // Build XLSX workbook from accumulated rows
-            let workbook_data =
-                XlsxWorksheetData { sheet_name: Some(request.table_name.clone()), columns: col_names, rows: all_rows };
-            let xlsx_bytes = build_xlsx_workbook(&workbook_data)?;
-            file.write_all(&xlsx_bytes).map_err(|e| format!("Failed to write XLSX file: {e}"))?;
+            finish_streaming_xlsx_workbook(writer).map_err(|e| format!("Failed to finalize XLSX file: {e}"))?;
         }
         "json" => {
             file.write_all(b"[\n").map_err(|e| format!("Failed to write JSON: {e}"))?;
@@ -636,6 +640,7 @@ pub async fn export_table_data_core(
 mod tests {
     use super::*;
     use crate::database_export::{clear_export_cancelled, set_export_cancelled};
+    use crate::xlsx_export::{build_xlsx_workbook, XlsxWorksheetData};
     use serde_json::json;
 
     // -----------------------------------------------------------------------

--- a/crates/dbx-core/src/xlsx_export.rs
+++ b/crates/dbx-core/src/xlsx_export.rs
@@ -1,6 +1,6 @@
 use serde::{Deserialize, Serialize};
 use serde_json::Value;
-use std::io::{Cursor, Write};
+use std::io::{Cursor, Seek, Write};
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
@@ -8,6 +8,131 @@ pub struct XlsxWorksheetData {
     pub sheet_name: Option<String>,
     pub columns: Vec<String>,
     pub rows: Vec<Vec<Value>>,
+}
+
+/// Streaming XLSX writer that incrementally writes rows to a ZIP-backed
+/// workbook.  This avoids accumulating all rows in memory before building the
+/// final file, drastically reducing peak memory for large exports.
+pub struct StreamingXlsxWriter<W: Write + Seek> {
+    zip: zip::ZipWriter<W>,
+    columns: Vec<String>,
+    next_row_number: usize,
+}
+
+/// Estimate column widths from header names only (used by the streaming path
+/// where full row data is not available up-front).  Each width is clamped to
+/// [10, 60] to stay within reasonable bounds.
+fn estimate_header_widths(columns: &[String]) -> Vec<usize> {
+    columns.iter().map(|col| (col.chars().count() + 2).clamp(10, 60)).collect()
+}
+
+/// Build the `<cols>` XML fragment from a width slice.
+fn cols_xml(widths: &[usize]) -> String {
+    widths
+        .iter()
+        .enumerate()
+        .map(|(index, width)| {
+            format!("<col min=\"{}\" max=\"{}\" width=\"{}\" customWidth=\"1\"/>", index + 1, index + 1, width)
+        })
+        .collect()
+}
+
+/// Build a single `<row>` XML fragment for the header row (row 1).
+pub(crate) fn header_row_xml(columns: &[String]) -> String {
+    format!(
+        "<row r=\"1\">{}</row>",
+        columns
+            .iter()
+            .enumerate()
+            .map(|(index, col)| cell_xml(Some(&Value::String(col.clone())), 0, index, Some(1)))
+            .collect::<String>()
+    )
+}
+
+/// Build a single `<row>` XML fragment for a data row.
+pub(crate) fn data_row_xml(row_number: usize, columns: &[String], row: &[Value]) -> String {
+    let cells = columns
+        .iter()
+        .enumerate()
+        .map(|(col_index, _)| cell_xml(row.get(col_index), row_number - 1, col_index, None))
+        .collect::<String>();
+    format!("<row r=\"{row_number}\">{cells}</row>")
+}
+
+fn write_zip_entry<W: Write + Seek>(zip: &mut zip::ZipWriter<W>, path: &str, content: &str) -> Result<(), String> {
+    let options = zip::write::SimpleFileOptions::default().compression_method(zip::CompressionMethod::Stored);
+    zip.start_file(path, options).map_err(|err| err.to_string())?;
+    zip.write_all(content.as_bytes()).map_err(|err| err.to_string())
+}
+
+/// Start a new streaming XLSX workbook.  The ZIP skeleton, worksheet header,
+/// column widths (estimated from header names) and the header row are written
+/// immediately.  Callers then feed data rows via [`StreamingXlsxWriter::write_row`]
+/// and finalize with [`StreamingXlsxWriter::finish`].
+pub(crate) fn start_streaming_xlsx_workbook<W: Write + Seek>(
+    writer: W,
+    sheet_name: Option<&str>,
+    columns: &[String],
+) -> Result<StreamingXlsxWriter<W>, String> {
+    let sheet_name = normalize_sheet_name(sheet_name);
+    let widths = estimate_header_widths(columns);
+
+    let mut zip = zip::ZipWriter::new(writer);
+    write_zip_entry(&mut zip, "[Content_Types].xml", content_types_xml())?;
+    write_zip_entry(&mut zip, "_rels/.rels", root_rels_xml())?;
+    write_zip_entry(&mut zip, "xl/workbook.xml", &workbook_xml(&sheet_name))?;
+    write_zip_entry(&mut zip, "xl/_rels/workbook.xml.rels", workbook_rels_xml())?;
+    write_zip_entry(&mut zip, "xl/styles.xml", styles_xml())?;
+
+    // Begin the sheet1.xml entry with header, frozen pane, column widths and
+    // the header row.
+    let options = zip::write::SimpleFileOptions::default().compression_method(zip::CompressionMethod::Stored);
+    zip.start_file("xl/worksheets/sheet1.xml", options).map_err(|err| err.to_string())?;
+
+    let sheet_header = format!(
+        concat!(
+            "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>",
+            "<worksheet xmlns=\"http://schemas.openxmlformats.org/spreadsheetml/2006/main\">",
+            "<sheetViews><sheetView workbookViewId=\"0\">",
+            "<pane ySplit=\"1\" topLeftCell=\"A2\" activePane=\"bottomLeft\" state=\"frozen\"/>",
+            "</sheetView></sheetViews>",
+            "<sheetFormatPr defaultRowHeight=\"15\"/>",
+            "<cols>{cols}</cols>",
+            "<sheetData>"
+        ),
+        cols = cols_xml(&widths),
+    );
+    zip.write_all(sheet_header.as_bytes()).map_err(|err| err.to_string())?;
+    zip.write_all(header_row_xml(columns).as_bytes()).map_err(|err| err.to_string())?;
+
+    Ok(StreamingXlsxWriter { zip, columns: columns.to_vec(), next_row_number: 2 })
+}
+
+impl<W: Write + Seek> StreamingXlsxWriter<W> {
+    /// Append a single data row to the worksheet.
+    pub fn write_row(&mut self, row: &[Value]) -> Result<(), String> {
+        self.zip
+            .write_all(data_row_xml(self.next_row_number, &self.columns, row).as_bytes())
+            .map_err(|err| err.to_string())?;
+        self.next_row_number += 1;
+        Ok(())
+    }
+
+    /// Finalize the worksheet and close the ZIP archive.  Returns the
+    /// underlying writer so callers can flush / close it as needed.
+    pub fn finish(mut self) -> Result<W, String> {
+        let row_count = self.next_row_number.saturating_sub(1);
+        let range = sheet_range(self.columns.len(), row_count);
+        self.zip
+            .write_all(format!("</sheetData><autoFilter ref=\"{range}\"/></worksheet>").as_bytes())
+            .map_err(|err| err.to_string())?;
+        self.zip.finish().map_err(|err| err.to_string())
+    }
+}
+
+/// Convenience wrapper that finalizes a streaming workbook.
+pub(crate) fn finish_streaming_xlsx_workbook<W: Write + Seek>(writer: StreamingXlsxWriter<W>) -> Result<W, String> {
+    writer.finish()
 }
 
 fn escape_xml(value: &str) -> String {
@@ -36,11 +161,7 @@ fn column_name(index: usize) -> String {
         out.push((b'A' + rem as u8) as char);
         n = (n - 1) / 26;
     }
-    // Safety: all pushed chars are ASCII, so reversing by bytes is correct
-    unsafe {
-        out.as_mut_vec().reverse();
-    }
-    out
+    out.chars().rev().collect()
 }
 
 fn cell_ref(row_index: usize, col_index: usize) -> String {
@@ -262,8 +383,10 @@ pub fn build_xlsx_workbook(data: &XlsxWorksheetData) -> Result<Vec<u8>, String> 
 
 #[cfg(test)]
 mod tests {
-    use super::{build_xlsx_workbook, XlsxWorksheetData};
+    use super::{build_xlsx_workbook, start_streaming_xlsx_workbook, XlsxWorksheetData};
+    use calamine::{open_workbook_auto, Reader};
     use serde_json::json;
+    use std::fs;
 
     #[test]
     fn builds_xlsx_zip_with_sheet_data() {
@@ -295,5 +418,26 @@ mod tests {
         .expect("build workbook");
         let text = String::from_utf8_lossy(&workbook);
         assert!(text.contains("name=\"bad name with chars and-a-very-\""));
+    }
+
+    #[test]
+    fn streams_xlsx_rows_to_a_readable_workbook() {
+        let path = std::env::temp_dir().join(format!("dbx-stream-test-{}.xlsx", uuid::Uuid::new_v4()));
+        {
+            let file = fs::File::create(&path).expect("create temp xlsx");
+            let mut writer =
+                start_streaming_xlsx_workbook(file, Some("Streamed"), &["id".to_string(), "name".to_string()])
+                    .expect("start workbook");
+            writer.write_row(&[json!(1), json!("Ada")]).expect("write row");
+            writer.write_row(&[json!(2), json!("Bob")]).expect("write row");
+            drop(writer.finish().expect("finish workbook"));
+        }
+
+        let mut workbook = open_workbook_auto(&path).expect("open workbook");
+        let range = workbook.worksheet_range("Streamed").expect("read worksheet");
+        assert_eq!(range.get_value((0, 0)).expect("header"), &calamine::Data::String("id".to_string()));
+        assert_eq!(range.get_value((1, 0)).expect("row1"), &calamine::Data::Float(1.0));
+        assert_eq!(range.get_value((2, 1)).expect("row2"), &calamine::Data::String("Bob".to_string()));
+        let _ = fs::remove_file(&path);
     }
 }

--- a/crates/dbx-core/tests/database_capabilities.rs
+++ b/crates/dbx-core/tests/database_capabilities.rs
@@ -188,6 +188,7 @@ fn skips_tcp_probe_for_local_file_plugin_and_agent_types() {
     assert!(skips_tcp_probe(&DatabaseType::Gbase));
     assert!(skips_tcp_probe(&DatabaseType::Databend));
     assert!(skips_tcp_probe(&DatabaseType::InfluxDb));
+    assert!(skips_tcp_probe(&DatabaseType::MessageQueue));
     assert!(!skips_tcp_probe(&DatabaseType::Postgres));
     assert!(!skips_tcp_probe(&DatabaseType::Mysql));
     assert!(!skips_tcp_probe(&DatabaseType::Gaussdb));

